src/shared/Database/SqlOperations.cpp

```diff
@@ -161,18 +160,26 @@ void SqlResultQueue::Update(uint32 timeout)
         threads[i] = new ACE_Based::Thread(caller);
     // Now execute thread unsafe callbacks
     MaNGOS::IQueryCallback* s = NULL;
-    uint32 unsafeQueryTime = WorldTimer::getMSTime();
+    uint32 begin = WorldTimer::getMSTime();
     while (_threadUnsafeWaitingQueries.next(s))
     {
         s->Execute();
         delete s;
         --numUnsafeQueries;
-        if (timeout && WorldTimer::getMSTimeDiffToNow(begin) > timeout)
-            break;
-    }
-    if (unsafeQueryTime > timeout)
-    {
-        sLog.out(LOG_PERFORMANCE, "Unsafe queries took longer than the timeout. %u remaining", numUnsafeQueries);
+
+        if (timeout != 0)
+        {
+            uint32 unsafeQueryTime = WorldTimer::getMSTimeDiffToNow(begin);
+            if (unsafeQueryTime > timeout)
```

Comment:
This is only going to break if the unsafe query logging threshold is greater than the timeout, defeating the purpose of performance logging.

The loop should always break if the timeout is reached, but only perf log if the unsafe logging threshold is exceeded. 

The actual issue is that the condition to log is checking against the timeout (static) rather than the running duration of the function, which has actually been fixed in another branch but not backported.